File: result/translate/starchat/java2cpp/c++/CHECK_LARGE_NUMBER_DIVISIBLE_11_NOT.cpp
bool f_gold ( string str ) {
        int n = str. length ( );
        int oddDigSum = 0, evenDigSum = 0;
        for ( int i = 0; i < n; i++ ) {
            if ( i % 2 == 0 ) oddDigSum += ( str[i] - '0' );
            else evenDigSum += ( str[i] - '0' );
        }
        return ( ( oddDigSum - evenDigSum ) % 11 == 0 );
    }

File: result/translate/transcoder/python2java/java/CHECK_LARGE_NUMBER_DIVISIBLE_11_NOT.java
public static boolean f_gold ( String str ) {
  int n = str . length ( ) ;
  int oddDigSum = 0 ;
  int evenDigSum = 0 ;
  for ( int i = 0 ;
  i <= n ;
  i ++ ) {
    if ( ( i % 2 == 0 ) && ( i % 3 == 0 ) ) oddDigSum = oddDigSum + ( Character . digit ( str . charAt ( i ) , 16 ) - 48 ) ;
    else evenDigSum = evenDigSum + ( Character . digit ( str . charAt ( i ) , 16 ) - 48 ) ;
  }
  return ( ( oddDigSum - evenDigSum ) % 11 == 0 ) ;
}